apps/prairielearn/src/pages/instructorLlmGenerateQuestion/instructorLlmGenerateQuestion.ts

```diff
@@ -0,0 +1,256 @@
+import * as path from 'path';
+
+import * as express from 'express';
+import asyncHandler from 'express-async-handler';
+import fs from 'fs-extra';
+import klaw from 'klaw';
+import { OpenAI } from 'openai';
+import { z } from 'zod';
+
+import * as error from '@prairielearn/error';
+import { loadSqlEquiv, queryRows, queryOptionalRow } from '@prairielearn/postgres';
+
+import { config } from '../../lib/config.js';
+import { IdSchema } from '../../lib/db-types.js';
+import { features } from '../../lib/features/index.js';
+import { REPOSITORY_ROOT_PATH } from '../../lib/paths.js';
+import { ServerJob, createServerJob } from '../../lib/server-jobs.js';
+
+import { DocumentChunk, buildContextForElementDocs } from './context-parsers/documentation.js';
+import { buildContextForQuestion } from './context-parsers/template-questions.js';
+import { AIGeneratePage } from './instructorLlmGenerateQuestion.html.js';
+
+const router = express.Router();
+
+const sql = loadSqlEquiv(import.meta.url);
+
+const QuestionGenerationContextEmbeddingSchema = z.object({
```

Comment:
Shift to `db-types.ts`?

Reply:
Done in d22c10e9a7f48ec19e32b8ab7142451b503b3c3b.